### PR TITLE
feat(platforms)!: make awscdk own package

### PR DESCRIPTION
**Pull request by hasanaburayyan**
## New Features
- Use custom platforms installed from npm (e.g. wing compile -t @winglang/platform-awscdk app.main.w)

## BREAKING CHANGES
- `awscdk` no longer a builtin platform, rather it is published as an external platform

## TODO:
- [x] use `tsup`
- [x] Fix e2e when in CI mode. (I broke plain old js files)

Closes: https://github.com/winglang/wing/issues/4693
## Checklist

- [x] Title matches [Winglang's style guide](https://www.winglang.io/contributing/start-here/pull_requests#how-are-pull-request-titles-formatted)
- [x] Description explains motivation and solution
- [x] Tests added (always)
- [x] Docs updated (only required for features)
- [x] Added `pr/e2e-full` label if this feature requires end-to-end testing

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Wing Cloud Contribution License](https://github.com/winglang/wing/blob/main/CONTRIBUTION_LICENSE.md)*.

### CODE REVIEW

File: libs/wingsdk/src/core/app.ts [hunk 1]
```diff
@@ -96,8 +96,8 @@ export abstract class App extends Construct {
    * Returns the root app.
    */
   public static of(scope: Construct): App {
-    if (scope instanceof App) {
-      return scope;
+    if ((scope as any)._isApp) {
+      return scope as any;
```
Comment (hasanaburayyan) [inline libs/wingsdk/src/core/app.ts:100]: The `instanceof` was misbehaving with external platform App instances which load in their own `@winglang/sdk`. I wrestled around with sandboxing but could not get things reasonably working.

Happy for other suggestions

File: libs/awscdk/package.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    "@winglang/sdk": "workspace:^"
+  },
+  "devDependencies": {
+    "bump-pack": "workspace:^",
+    "typescript": "^4.9.5",
+    "tsup": "^6.7.0",
+    "vitest": "^0.32.4"
+  },
+  "scripts": {
+    "compile": "tsc",
```
Comment (hasanaburayyan) [inline libs/awscdk/package.json:25]: still trying to get `tsup` set up, had some issues but didnt want to hold up initial reviews while I play with this.

**Review by eladb — APPROVED** [review]
Ship it!

(make sure to update docs!)

File: libs/awscdk/turbo.json [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        "dist/**",
+        "lib/**"
+      ]
+    },
+    "test": {
+      "dependsOn": ["compile"]
+    },
+    "package": {
+      "dependsOn": ["compile"],
+      "outputs": ["../../dist/winglang/platform-awscdk-*.tgz"]
```
Comment (MarkMcCulloh) [inline libs/awscdk/turbo.json:18]:
```suggestion
      "outputs": ["../../dist/winglang-platform-awscdk-*.tgz"]
```

**Pull request merged**